### PR TITLE
Adjust bounding box of data operation whenever its description is changed

### PR DESCRIPTION
**Pull request by wdednam**
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/132)

### CODE REVIEW

File: model/operation.cc [hunk 1]
```diff
@@ -678,6 +678,18 @@ namespace minsky
...
+  
+  string DataOp::description_() const
+  {
+	return m_description;  
+  }
+   
+  string DataOp::description_(const std::string& x)
+  {
+    bb.update(*this); // adjust icon bounding box - see ticket #704
+    m_description=x;
```
Comment (highperformancecoder) [inline model/operation.cc]: These two statements are done in the wrong order

File: model/operation.cc [hunk 2]
```diff
@@ -694,8 +706,8 @@ namespace minsky
     size_t p=fileName.rfind('/');
     // '/' is guaranteed not to be in fileName, so we can use that as
     // a delimiter
-    description = "\\verb/"+
-      ((p!=string::npos)? fileName.substr(p+1): fileName) + "/";
+    description("\\verb/"+
+      ((p!=string::npos)? fileName.substr(p+1): fileName) + "/");
     //initXVector();
     bb.update(*this); // adjust bounding box for data import operation - see ticket #1121
```
Comment (highperformancecoder) [inline model/operation.cc]: Can be removed now

File: model/operation.h [hunk 3]
```diff
@@ -206,8 +206,27 @@ namespace minsky
   class DataOp: public ItemT<DataOp, Operation<minsky::OperationType::data>>
   {
     CLASSDESC_ACCESS(DataOp);
+    friend struct SchemaHelper;
+    string m_description;
   public:
-    string description;
+    DataOp() {description("");}
```
Comment (highperformancecoder) [inline model/operation.h]: The default default constructor would be fine too (bb is updated next time it is used).

File: model/operation.h [hunk 4]
```diff
@@ -206,8 +206,27 @@ namespace minsky
...
+    // ensure that copies create a new data operation
+    DataOp(const DataOp& x): 
+      OperationBase(x){group.reset(); description(x.description());}
```
Comment (highperformancecoder) [inline model/operation.h]: Whilst probably harmless, I was wondering why you added group.reset() here. group.reset is already done in ItemT::clone. None of the other item copy constructors have this, except for IntOp. For consistency, I suggest removing the group.reset here and in IntOp. You can log this as a low priority backlog ticket, however, if your'e concerned we might not understand the implications of that change.

File: model/operation.cc (hunk 1 above)
Comment (wdednam) [reply model/operation.cc]: Thanks for catching this.

File: model/operation.cc (hunk 2 above)
Comment (wdednam) [reply model/operation.cc]: I have done so, thanks.

File: model/operation.h (hunk 3 above)
Comment (wdednam) [reply model/operation.h]: I have also made this the default constructor.

File: model/operation.h (hunk 4 above)
Comment (wdednam) [reply model/operation.h]: Sorry, I missed this and will commit the changes after the current Travis build has finished. It didn't occur to me that the group.reset() in ItemT:clone() would work for these objects too. I sometimes forget that these and other objects fall under the item object umbrella.

**Pull request merged**